core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheManager.java

```diff
@@ -531,6 +531,23 @@ public State state() {
     return mState.get();
   }
 
+  @Override
+  public boolean append(PageId pageId, int appendAt, byte[] page, CacheContext cacheContext) {
+    LOG.debug("append ({},{} bytes) enters", pageId, page.length);
+    if (mState.get() != READ_WRITE) {
+      Metrics.PUT_NOT_READY_ERRORS.inc();
+      Metrics.PUT_ERRORS.inc();
+      return false;
+    }
+    byte[] newPage = new byte[appendAt + page.length];
```

Comment:
Will these be frequent operations? Does it make sense to pre-allocate some buffers rather than allocate new every time?

Reply:
good call, I refined the code, the new logic would only pre-allocate buffers when necessary